src/graph.rs

```diff
@@ -0,0 +1,106 @@
+use std::collections::HashMap;
+
+use petgraph::graphmap::UnGraphMap;
+
+use crate::moves::Direction;
+use crate::{Board, Coord};
+
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub enum SpaceType {
+    Empty,
+    Food,
+    Hazard,
+    Occupied,
```

Comment:
I think that eventually we will want to break this down by snake (especially for certain game modes)

Reply:
yeah but for A* it's just pathfinding between a source and destination, I think if we want to have more complicated logic here we might not just want a shortest path algorithm